### ISSUE TITLE
Introduce the new validation graph widget

### ISSUE THREAD

**Issue by hgonomeg** [hgonomeg/coot#33]
The new validation graph widget (being developed under #31) needs to be plugged into the codebase.

## Possible approaches

1. Use normal menu items powered by a collection of `GMenu`s that get updated whenever a model gets added/deleted.
2. Build a new dialog ("Show validation graph" or something like that) with two comboboxes: one for choosing the graph type and another one to choose the model.
3. Create a special expandable/resizable pane for managing validation graphs. It would have a combobox to choose the model and a grid of checkbuttons to select which graph types should be displayed.

**Third is what we go with.**

# Tasks

* [x] Choose approach for replacing `dynamic-menus`
* [ ] Unify representation of graph types (ideally single enum, no strings [or a single string-to-enum function])
  * [x] Add new enum
  * [ ] Possibly remove the old one
* [x] Implement the chosen replacement of `dynamic-menus`
  * [x] Add the validation graphs pane
  * [x] Add the internal layout of the pane - combobox and space for checkbuttons
  * [x] Implement choosing models from the combobox
  * [x] Add check buttons
  * [x] Implement the check buttons' logic
  * [x] Remove the validation graph-related menu items from the UI
* [ ] Remove `dynamic-menus.{cc,hh}`
* [ ] Add the graph widget code to `coot` directly [copy from `test-validation/` (preferably after #31 is closed)]
* [ ] Rework graph presentation - remove the dialog and make an overlay with tabs
  * [x] Add the overlay to the UI
  * [x] Implement tab management
  * [ ] Implement showing the graph widget in tabs' content
  * [ ] Remove the old dialog
* [ ] Rework how graphs are stored and managed in `graphics_info_t`
  * [x] Use `std::map` for storing any widgets/graphs.
  * [ ] Obtain + use a new unified graph-type-agnostic graph data representation type
    * [ ] Adjust the widget to the requirements
  * [x] Use `std::map` to store graph data
  * [ ] Adapt code responsible for computing validation data for all graph types
  * [x] Rewrite validation graph creation
  * [ ] Rewrite `update_validation_graphs()`
  * [ ] Clean up
  * [x] Adjust after receiving feedback
    * [x] Make the control UI a pop-up dialog, not a pane
    * [x] Remove the tabs and add some form of chain-selection [combobox?]
* [ ] Test thoroughly

**Comment by hgonomeg** [comment]
UI design illustration:

![image_tabs](https://user-images.githubusercontent.com/48153724/198635063-ffbf6438-972d-40c8-a8db-b8d40645d9bb.png)
![Screenshot from 2022-10-28 13-38-29](https://user-images.githubusercontent.com/48153724/198635135-d1826160-3270-45b3-93e8-bc69f6a44de5.png)

**Comment by hgonomeg** [comment]
![new_graph_management_design](https://user-images.githubusercontent.com/48153724/198681270-096e67d4-db59-4824-991c-df040985f11c.png)
` update_validation_graphs()` corresponds to `on_graphed_model_updated()`

**Comment by hgonomeg** [comment]
Progress demo:
![image](https://user-images.githubusercontent.com/48153724/199804914-7f6043a0-1a28-41c9-8e50-86c21f4e2103.png)

**Comment by hgonomeg** [comment]
#34 got merged into the `validation-graphs` working branch.
This issue can now be renamed from "Prepare for introducing the new validation graph widget" into "Introduce the new validation graph widget"
The `validation-graphs` branch is from now on the common development branch for both #33 and #31

**Comment by hgonomeg** [comment]
Updated:
![new_graph_management_design2](https://user-images.githubusercontent.com/48153724/203717624-0f281648-112a-4137-bfd8-03e5a0b20e89.png)

**Comment by hgonomeg** [comment]
Validation graphs have been introduced by Paul and this issue can be considered closed. 
Remaining cleanup work can be continued here: #40
